site/src/app/v2/trouver-un-club/agent.ts

```diff
@@ -1,13 +1,15 @@
 export interface SqlSearchParams {
   nom?: string;
+  limit?: number;
+  offset?: number;
 }
 
-export const getClubs = async (param?: SqlSearchParams): Promise<SportGouvJSONResponse> => {
-  const queryString: URLSearchParams = new URLSearchParams('nom%20is%20not%20null&limit=20');
+export const getClubs = async (param: SqlSearchParams): Promise<SportGouvJSONResponse> => {
+  const queryString: URLSearchParams = new URLSearchParams();
 
-  if (param?.nom) {
-    queryString.append('where', param.nom);
-  }
+  queryString.append('limit', param.limit ? param.limit.toString() : '20');
+  queryString.append('offset', param.offset ? param.offset.toString() : '0');
+  queryString.append('where', `nom is not null ${param?.nom ? `AND ${param.nom}` : ''} `);
```

Comment:
Si tu veux être plus clean, on pourrait utiliser la Web API built-in d'URL de la manière suivante:

```suggestion
  const baseUrl =
    'https://sports-sgsocialgouv.opendatasoft.com/api/explore/v2.1/catalog/datasets/passsports-asso_volontaires/records';

  const url = new URL(baseUrl);

  const params: URLSearchParams = new URLSearchParams();

  params.append('limit', param.limit ? param.limit.toString() : '20');
  params.append('offset', param.offset ? param.offset.toString() : '0');
  params.append('where', `nom is not null ${param?.nom ? `AND ${param.nom}` : ''} `);

  url.search = params.toString();

  const response = await fetch(url);
```

Reply:
j'ai pris ta version plus clean. J'ai fais une modif mineure sur le where qui avait des espaces inutiles.

Reply:
j'ai ajouté commune et adresse dans la clé (en plus du nom). Un doublon peu théoriquement exister mais c'est peu probable. Il n'y a pas de champ unique pour les clubs à priori.

Reply:
Parfait 👍